File: src/main/java/com/itmo/microservices/users/controller/UserController.java
package com.itmo.microservices.users.controller;

import lombok.AllArgsConstructor;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;
import com.itmo.microservices.users.entity.AppUser;
import com.itmo.microservices.users.model.AppUserDto;
import com.itmo.microservices.users.model.RegistrationRequest;
import com.itmo.microservices.users.service.UserService;

@RestController
@RequestMapping("/users")
@AllArgsConstructor
public class UserController {

    private final UserService userService;

    @PostMapping
    public void register(@RequestBody RegistrationRequest request) {

    }

    @GetMapping("/me")
    public AppUserDto getCurrentUserInfo(@AuthenticationPrincipal AppUser user) {
        return null;
    }

    @DeleteMapping("/me")
    public void deleteCurrentUser(@AuthenticationPrincipal AppUser user) {

    }
}

File: src/main/kotlin/com/itmo/microservices/demo/users/controller/UserController.kt
package com.itmo.microservices.demo.users.controller

import com.itmo.microservices.demo.users.entity.AppUser
import com.itmo.microservices.demo.users.model.AppUserDto
import com.itmo.microservices.demo.users.model.RegistrationRequest
import com.itmo.microservices.demo.users.service.UserService
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/users")
class UserController(private val userService: UserService) {

    @PostMapping
    fun register(@RequestBody request: RegistrationRequest) {
    }

    @GetMapping("/me")
    fun getCurrentUserInfo(@AuthenticationPrincipal user: AppUser): AppUserDto? = null

    @DeleteMapping("/me")
    fun deleteCurrentUser(@AuthenticationPrincipal user: AppUser) = Unit
}